C#/C#.sublime-syntax

```diff
@@ -1,278 +1,1634 @@
+
+  preprocessor_option:
+    - meta_scope: meta.preprocessor.cs
+    - match: '(define|undef)\s+{{name}}'
+      captures:
+        1: keyword.control.preprocessor.cs
+        2: constant.numeric.cs
+    - match: '((el)?if)'
+      scope: keyword.control.preprocessor.cs
       push:
-        - meta_scope: meta.block.source.cs
-        - match: "}"
-          scope: punctuation.section.block.end.source.cs
-          pop: true
-        - include: code
-  builtinTypes:
-    - match: \b(bool|byte|sbyte|char|decimal|double|float|int|uint|long|ulong|object|short|ushort|string|void|class|struct|enum|interface)\b
-      scope: storage.type.source.cs
-  class:
-    - match: '(?=\w?[\w\s]*(?:class|struct|interface|enum)\s+\w+)'
+        - match: '\(|\)'
+          scope: punctuation.definition.expression.cs
+        - match: '&&|\|\||!'
+          scope: keyword.operator.logical.cs
+        - match: '\b(true|false)\b'
+          scope: constant.language.cs
+        - match: '{{name}}'
+          scope: constant.numeric.cs
+        - include: option_done
+    - match: '(else|endif)'
+      scope: keyword.control.preprocessor.cs
+      push: option_done
+    - match: '(error|warning)\s*'
+      scope: keyword.other.preprocessor.cs
+      push: option_message
+    - match: '(region)\s*'
+      scope: storage.type.section.cs
       push:
-        - meta_scope: meta.class.source.cs
-        - match: "}"
-          scope: punctuation.section.class.end.source.cs
-          pop: true
-        - include: storage-modifiers
-        - include: comments
-        - match: (class|struct|interface|enum)\s+(\w+)
-          scope: meta.class.identifier.source.cs
+        - match: ([^/\s][^/]*?)\s*(?=//|/\*|$)
           captures:
-            1: storage.modifier.source.cs
-            2: entity.name.type.class.source.cs
-        - match: ":"
-          push:
-            - match: "(?={)"
-              pop: true
-            - match: '\s*,?([A-Za-z_]\w*)\b'
-              captures:
-                1: storage.type.source.cs
-        - match: "{"
-          scope: punctuation.section.class.begin.source.cs
-          push:
-            - meta_scope: meta.class.body.source.cs
-            - match: "(?=})"
+            1: entity.name.section.cs
+        - include: option_done
+    - match: '(endregion)\s*'
+      scope: storage.type.section.cs
+      push:
+        - match: ([^/\s][^/]*?)\s*(?=//|/\*|$)
+          captures:
+            1: variable.other.section.cs
+        - include: option_done
+
+    - match: '(line)\s+(default|hidden)'
+      captures:
+        1: keyword.other.preprocessor.cs
+        2: keyword.other.preprocessor.cs
+      push: option_done
+    - match: '(line)\s*([0-9]*)\s*(".*")?'
+      captures:
+        1: keyword.other.preprocessor.cs
+        2: constant.numeric.cs
+        3: string.quoted.double.cs
+      push: option_done
+    - match: '(pragma\s+checksum)\s+(".*")\s+("\{)(.*)(\}")\s+("\{)(.*)(\}")'
```

Comment:
Add a `\b` at the beginning.

The regex is getting really complicated. If possible, push into a context that handles the strings instead of trying to match in a single regex. You can exit the context as soon as something isn't valid syntax.